src/main/java/net/smartcosmos/userdetails/autoconfigure/UserDetailsAutoConfiguration.java

```diff
@@ -0,0 +1,60 @@
+package net.smartcosmos.userdetails.autoconfigure;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.validation.beanvalidation.LocalValidatorFactoryBean;
+
+import net.smartcosmos.userdetails.service.AuthenticateUserService;
+import net.smartcosmos.userdetails.service.AuthenticateUserServiceDefault;
+import net.smartcosmos.userdetails.service.UserDetailsService;
+
+/**
+ * Auto-configuration that creates default beans if no other versions exist.
+ */
+@Configuration
+@ComponentScan(basePackages = "net.smartcosmos.userdetails")
+public class UserDetailsAutoConfiguration {
+
+    /**
+     * If no custom implementation of {@link AuthenticateUserService} is available, use the default.
+     *
+     * @param userDetailsService the user details service
+     * @return the AuthenticateUserService bean
+     */
+    @Bean
+    @ConditionalOnMissingBean
```

Comment:
If there is no value in the annotation, it basically says _use me if you don't have anything of my type_.

I'll add the log message.